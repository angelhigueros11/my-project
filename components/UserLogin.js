import React, { useState } from "react";
import {
  StyleSheet,
  Text,
  View,
  SafeAreaView,
  TextInput,
  TouchableOpacity,
  Pressable,
  ActivityIndicator,
} from "react-native";
import { MaterialCommunityIcons } from "@expo/vector-icons";
import { useTogglePasswordVisibility } from "./hooks/useTogglePasswordVisibility";
import { auth } from "./firebase.js";

export default function UserLogin({ navigation }) {
  const { passwordVisibility, rightIcon, handlePasswordVisibility } =
    useTogglePasswordVisibility();
  const [password, setPassword] = useState("");
  const [username, setUsername] = useState("");
  const [mailSubmitting, setMailSubmitting] = useState(false);

  const handleLogin = () => {
    setMailSubmitting(true);
    auth
      .signInWithEmailAndPassword(username.trim(), password)
      .then((userCredentials) => {
        const user = userCredentials.user;
        let name = username.split(".")[0];
        name = name.split("@")[0];
        name = name.charAt(0).toUpperCase() + name.slice(1);
        let email = username;
        setMailSubmitting(false);
        navigation.navigate("Home", { email, name });
      })
      .catch((err) => {
        setMailSubmitting(false);
        alert(err.message);
      });
  };

  return (
    <SafeAreaView style={styles.container}>
      {/* LOGO AREA */}
      <View style={styles.logo}>
        <Text style={styles.logoText1}>Health</Text>
        <Text style={styles.logoText2}>Forms</Text>
      </View>

      {/** Area de Inputs */}
      <View style={styles.containerInputs}>
        <TextInput
          style={styles.input}
          placeholder="Correo"
          value={username}
          onChangeText={(text) => setUsername(text)}
          autoCapitalize="none"
          placeholderTextColor="#fff"
        />

        {/* Area del input contrasena */}
        <View style={styles.inputContainer}>
          <TextInput
            style={styles.inputField}
            name="password"
            placeholder="Contraseña"
            autoCapitalize="none"
            autoCorrect={false}
            textContentType="newPassword"
            secureTextEntry={passwordVisibility}
            value={password}
            enablesReturnKeyAutomatically
            onChangeText={(text) => setPassword(text)}
            placeholderTextColor="#fff"
          />

          {/* Area del icono */}
          <Pressable
            onPress={handlePasswordVisibility}
            style={{ top: 10, right: 20 }}
          >
            <MaterialCommunityIcons
              name={rightIcon}
              size={28}
              color="#f5f5f5"
            />
          </Pressable>
        </View>

        {/* Iniciar Sesion */}
        {!mailSubmitting && (
          <TouchableOpacity style={styles.logInButton} onPress={handleLogin}>
            <Text style={{ color: "white" }}>Iniciar Sesión</Text>
          </TouchableOpacity>
        )}

        {mailSubmitting && (
          <TouchableOpacity
            style={[
              styles.logInButton,
              {
                marginTop: 0,
                flexDirection: "row",
                justifyContent: "space-around",
              },
            ]}
            disabled={true}
          >
            <ActivityIndicator size={24} color="white" />
          </TouchableOpacity>
        )}

        {/* Crear Perfil */}
        <TouchableOpacity
          style={[
            styles.logInButton,
            { marginTop: 20, backgroundColor: "#143590" },
          ]}
          onPress={() => {
            navigation.navigate("CreateProfile");
          }}
        >
          <Text style={{ color: "#fff" }}>Crear perfil</Text>
        </TouchableOpacity>
      </View>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "flex-start",
    backgroundColor: "#143590",
  },
  logo: {
    flex: 0.3,
    flexDirection: "row",
    alignItems: "flex-end",
    justifyContent: "center",
  },
  logoImg: {
    resizeMode: "center",
    width: 250,
    height: 50,
    bottom: 0,
    backgroundColor: "#fff",
  },
  containerInputs: {
    flex: 0.5,
    backgroundColor: "#143590",
    width: "80%",
    alignItems: "center",
  },
  input: {
    backgroundColor: "#143590",
    marginTop: 50,
    height: 40,
    borderBottomColor: "#fff",
    borderBottomWidth: 1,
    width: "80%",
    color: "#fff",
  },
  logInButton: {
    width: "80%",
    height: 40,
    borderRadius: 15,
    backgroundColor: "#0657a2",
    justifyContent: "center",
    alignItems: "center",
    position: "relative",
    top: 60,
    color: "#fff",
  },

  inputContainer: {
    marginTop: 10,
    backgroundColor: "#143590",
    width: "80%",
    flexDirection: "row",
    alignItems: "center",
    borderColor: "white",
  },
  inputField: {
    backgroundColor: "#143590",
    marginTop: 20,
    height: 40,
    borderBottomColor: "#fff",
    borderBottomWidth: 1,
    width: "100%",
    color: "white",
  },
  googleButtonContainer: {
    backgroundColor: "red",
  },
  logoText1: {
    fontSize: 40,
    color: "#fff",
    fontWeight: "bold",
  },
  logoText2: {
    fontSize: 40,
    color: "#00c9cc",
    fontWeight: "bold",
  },
});
