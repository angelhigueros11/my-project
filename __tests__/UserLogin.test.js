import React from 'react';
import renderer from 'react-test-renderer';
import './Mocks/addEventListener.mock';
import UserLogin from '../components/UserLogin';

jest.mock("@react-navigation/native", () => {
    const actualNav = jest.requireActual("@react-navigation/native");
    return {
      ...actualNav,
      useNavigation: () => ({
        navigate: jest.fn(),
        dispatch: jest.fn(),
      }),
    };
});

describe('Pruebas en <UserLogin />', () => {
    it('Renders correctly without params', () => {
        const tree = renderer.create(<UserLogin />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Renders correctly with params', () => {
        const navigation = { navigate: jest.fn() };
        const tree = renderer.create(<UserLogin navigation={navigation}/>).toJSON();
        expect(tree).toMatchSnapshot();
    }); 
    
});