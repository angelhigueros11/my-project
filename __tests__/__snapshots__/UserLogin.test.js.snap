// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <UserLogin /> Renders correctly with params 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#143590",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flex": 0.3,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 40,
          "fontWeight": "bold",
        }
      }
    >
      Health
    </Text>
    <Text
      style={
        Object {
          "color": "#00c9cc",
          "fontSize": 40,
          "fontWeight": "bold",
        }
      }
    >
      Forms
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#143590",
        "flex": 0.5,
        "width": "80%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="Correo"
      placeholderTextColor="#fff"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "#143590",
          "borderBottomColor": "#fff",
          "borderBottomWidth": 1,
          "color": "#fff",
          "height": 40,
          "marginTop": 50,
          "width": "80%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#143590",
          "borderColor": "white",
          "flexDirection": "row",
          "marginTop": 10,
          "width": "80%",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        enablesReturnKeyAutomatically={true}
        name="password"
        onChangeText={[Function]}
        placeholder="Contraseña"
        placeholderTextColor="#fff"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "backgroundColor": "#143590",
            "borderBottomColor": "#fff",
            "borderBottomWidth": 1,
            "color": "white",
            "height": 40,
            "marginTop": 20,
            "width": "100%",
          }
        }
        textContentType="newPassword"
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "right": 20,
            "top": 10,
          }
        }
      >
        <Text />
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#0657a2",
          "borderRadius": 15,
          "color": "#fff",
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "position": "relative",
          "top": 60,
          "width": "80%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
          }
        }
      >
        Iniciar Sesión
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#143590",
          "borderRadius": 15,
          "color": "#fff",
          "height": 40,
          "justifyContent": "center",
          "marginTop": 20,
          "opacity": 1,
          "position": "relative",
          "top": 60,
          "width": "80%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
          }
        }
      >
        Crear perfil
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`Pruebas en <UserLogin /> Renders correctly without params 1`] = `
<View>
  <Text>
    ERROR
  </Text>
</View>
`;
